src/com/googlecode/shavenmaven/Resolver.java

```diff
@@ -47,17 +49,17 @@ public Resolver(File directory) {
         this(directory, System.out);
     }
 
-    public boolean resolve(Artifact artifact) throws Exception {
```

Comment:
Sorry, I'm obviously not familiar with TotallyLazy: does it provide a maybe monad like Optional?  That's probably the best approach.